Made UrlDetector work with color codes.

FireInstall opened the pull request:
Ich konnte es nicht auf mir sitzen lassen, dass ein Pattern besser sein soll, als der URL-detector.
Also funktioniert der Detektor jetzt auch mit Farbcodes, die man - im Gegensatz zum Pattern - an beliebiger stelle ohne Beschränkungen einfügen kann und trotzdem von der Vollständigkeit des Detectors profitiert.

Ja ich habe einen Schaden. Danke der Nachfrage.

FireInstall committed:
Made UrlDetector work with color codes.

use correct url-detector lib
update ProtocolLib, so it's compiling again.

FireInstall commented:
Ich habe leider auch keine Ahnung was GitHub für ein Problem mit dem Diff hat. Ich habe extra grade noch einmal geforcepusht um keine unnötigen white-space Änderungen drin zu haben. Es macht keinen Unterschied. Sind es line endings die sich ändern?!? Dachte die sind standart Linux, wenn Intellij die mit Git pust oder so

FireInstall committed:
add .iml to gitignore.
I don't know why intellij generates it in th first place, but all the methods of the bug tracker are outdated.

FireInstall commented:
forcepush für eclipse formatting. + .gitignore

Schpammer commented:
Ich habe den Code nur mal kurz überflogen. Wie verhält es sich, wenn du 2x den gleichen Link in der Nachricht hast? Und ich glaube ein paar Kommentare würden nicht schaden. Wir sollten das dann auch mit Joshis PR zusammenführen, damit der PR bei Venturechat auch auf Stand ist. Habe den anderen PR auch gesehen mit den Color Codes und dem Spaghetti-Regex :D

FireInstall commented:
Hier muss nichts zusammengeführt werden, diese PR liegt logisch als auch in der Commit-History nach Joshies Änderungen. Alles was Joshis PR macht ist hier auch drin, da seine Änderungen meine Grundlage waren. Wird diese PR gemerged geht es ab in Joshis branch und die PR bei Venture ist auf diesen Stand hier. Muss nur die Beschreibung dort gegebendenfalls angepasst werden.
Daher gilt deine Frage nach den mehrfachen Links uns beiden gleichermaßen. Ehrlich gesagt weiß ich nicht genau, wie der UrlDetector mit Dopplungen umgeht, aber gemessen daran, dass er eine Liste und kein Set zurück gibt nehme ich mal ganz stark an, daran wurde gedacht.
Mein Code vergleicht nun den gefundenen Link mit den original Text, findet die Unterschiede - aka die Formatierungscodes und fügt sie wieder ein. Also sollte alles Taco sein, was Dopplungen angeht.

Kommentare kann ich einfügen 👍

FireInstall committed:
Added way more comments to clarify what the code does
Fix wrongly overwriting every formatting code with the last one, even if they should work together like bold and underlined.
Note: As per Minecraft's custom colour codes overwrite all formatting!

FireInstall commented:
@Schpammer  Hoffe es ist so besser ersichtlich, was der Code macht!

FireInstall commented:
force push um commit message typo zu berichtigen

FireInstall commented:
> Ich habe leider auch keine Ahnung was GitHub für ein Problem mit dem Diff hat. Ich habe extra grade noch einmal geforcepusht um keine unnötigen white-space Änderungen drin zu haben. Es macht keinen Unterschied. Sind es line endings die sich ändern?!? Dachte die sind standart Linux, wenn Intellij die mit Git pust oder so

Es sind die LineEndings. Aus bisher unbekannten Gründen ist Format.java im Original CRLF, während Stichprobenartig der Rest LF zu nutzen scheint.
Ich sehe es mal als feature an, die LineEndings mit dem Rest des Repos in Standart zu bringen.
